### PR TITLE
Kernel loader implemented!

### LINKED ISSUE #16: Implement the kernel loader
Implement two functions, one to load the kernel and return it's base address, the other to get it's entry point.

### PR DESCRIPTION
**Pull request by ExhoAR22**
This closes #16.

### CODE REVIEW

File: bootloader/ELF.h [hunk 1]
```diff
@@ -0,0 +1,191 @@
+#pragma once
+
+// This header defines everything related to the ELF specification.
+
+namespace elf {
+    using int8   = char;
+    using uint8  = unsigned char;
+    using int16  = short;
+    using uint16 = unsigned short;
+    using int32  = long;
```
Comment (Itay2805) [inline bootloader/ELF.h]: If you are using gcc the long is like size_t, aka, 32bit when compiling for 32bit and 64bit when compiling for 64bit

File: bootloader/Utils.cpp [hunk 2]
```diff
@@ -20,3 +20,67 @@ uefi::Status utils::enterOptimalTextMode(const uefi::SystemTable* systemTable) {
...
+
+    uint8 data = 0;
+    uefi::protocols::FileProtocolInfo* infoBuffer = reinterpret_cast<uefi::protocols::FileProtocolInfo*>(&data);
+    uintn infoSize = 1;
+    status = file->getInfo(file, &uefi::protocols::FILE_INFO_GUID, &infoSize, infoBuffer);
+    if (status != uefi::Status::BUFFER_TOO_SMALL) {
+        return nullptr;
+    }
+
+    status = systemTable->bootServices->allocatePool(uefi::MemoryType::LOADER_DATA, infoSize, reinterpret_cast<void**>(&infoBuffer));
```
Comment (Itay2805) [inline bootloader/Utils.cpp]: You never free this buffer

Comment (ExhoAR22) [reply bootloader/Utils.cpp]: Sure I do, line 72.

File: bootloader/ELF.h (hunk 1 above)
Comment (ExhoAR22) [reply bootloader/ELF.h]: I'm using clang by definition, but it should be fixed.
What would you suggest?
```c++
#ifdef __GNUC__
...
#elif __clang__
...
#endif

File: bootloader/Utils.cpp (hunk 2 above)
Comment (Itay2805) [reply bootloader/Utils.cpp]: Woops, my bad.

File: bootloader/ELF.h (hunk 1 above)
Comment (ExhoAR22) [reply bootloader/ELF.h]: I see now that it won't be freed if a failure occurs.
I'll fix it soon.

Comment (Itay2805) [reply bootloader/ELF.h]: It is the same on clang (or any compiler on Linux) the way to handle it is probably to just use int. You can also add static asserts on sizes.

File: bootloader/Utils.cpp (hunk 2 above)
Comment (ExhoAR22) [reply bootloader/Utils.cpp]: I see now that it won't be freed if a failure occurs.
I'll fix it soon.

File: bootloader/ELF.h (hunk 1 above)
Comment (ExhoAR22) [reply bootloader/ELF.h]: Fixed!

File: bootloader/Utils.cpp (hunk 2 above)
Comment (ExhoAR22) [reply bootloader/Utils.cpp]: Fixed!

**Pull request merged**